src/core/mount-setup.c

```diff
@@ -229,76 +229,105 @@ int mount_setup_early(void) {
         return mount_points_setup(N_EARLY_MOUNT, false);
 }
 
-int mount_cgroup_controllers(char ***join_controllers) {
+static const char *join_with(const char *controller) {
+
+        static const char* const pairs[] = {
+                "cpu", "cpuacct",
+                "net_cls", "net_prio",
+                NULL
+        };
+
+        const char *const *x, *const *y;
+
+        assert(controller);
+
+        /* This will lookup which controller to mount another controller with. Input is a controller name, and output
+         * is the other controller name. The function works both ways: you can input one and get the other, and input
+         * the other to get the one. */
+
+        STRV_FOREACH_PAIR(x, y, pairs) {
+                if (streq(controller, *x))
+                        return *y;
+                if (streq(controller, *y))
+                        return *x;
+        }
+
+        return NULL;
```

Comment:
I like the nice and generic code, but since this is all hardcoded anyway, why not just write:
```C
{
       if (streq(c, "cpu")) return "cpuacct";
       if (streq(c, "cpuacct")) return "cpu";
       if (streq(c, "net_cls")) return "net_prio";
       if (streq(c, "net_prio")) return "net_cls";
       return NULL;
}
```
?

Reply:
i had it like this initially, but then rewrote it again, because it looked so redundant... maybe let's call this a matter of taste? ,-)

Reply:
OK ;)